libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -1583,5 +1583,9 @@ public void onGutenbergDialogNegativeClicked(@NonNull String instanceTag) {
     @Override
     public void onConnectionStatusChange(boolean isConnected) {
         getGutenbergContainerFragment().onConnectionStatusChange(isConnected);
+        if (hasFailedMediaUploads()) {
+            mEditorFragmentListener.onMediaRetryAll(mFailedMediaIds);
+            mFailedMediaIds.clear();
```

Comment:
Not sure clearing this is necessary, it does not appear to be done elsewhere, e.g., [`showRetryMediaUploadDialog`](https://github.com/wordpress-mobile/WordPress-Android/blob/f4bc1f62fe6b0be7f42ae4bbfed60f6931e5982a/libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java#L917-L944). This clear was added in an attempt to fix erroneously displayed errors after saving or publishing a post, but it does not appear to work. 

<details><summary>Error screenshot</summary>

![Unexpected failed media upload error after closing editor](https://github.com/wordpress-mobile/WordPress-Android/assets/438664/d7b59e78-cbf2-4e8a-a0f4-724d2ea06fc0)


</details>

It appears as though a lingering error state/message is in place, but I am unable to find where that is or how to clear it. I expected [`onMediaRetryAll`](https://github.com/wordpress-mobile/WordPress-Android/blob/f4bc1f62fe6b0be7f42ae4bbfed60f6931e5982a/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L3303-L3311) to clear it, but it doesn't appear to be doing so. 

👋🏻 @antonis. Randomly pinging you for help. 😄 Do you happen to know why I might see this error when the media attached to the post actually _successfully uploads_ with the new retry logic?

Reply:
After testing the existing functionality of retrying failing media uploads via the dialog presented when tapping a failed image, it appears the lingering error message is a preexisting issue. Semi-related issues are reported in https://github.com/wordpress-mobile/WordPress-Android/issues/11039, https://github.com/wordpress-mobile/WordPress-Android/issues/15897, and https://github.com/wordpress-mobile/WordPress-Android/issues/13427. 

<details><summary>Step to reproduce</summary>

1. Disable network connectivity.
2. Add an Image block and attach media. 
3. Enable network connectivity. 
4. Tap the Image block with a failed upload. 
5. Tap "Retry."
6. After the media finishes uploading successfully, save and close the post. 
7. Note the failed media upload notification. 

</details>

Therefore, I do not consider this issue blocking. It'd be nice to resolve, but could be done so in a separate PR later.

Reply:
> Not sure clearing this is necessary

Removing the clearing with c114528b47a56b3f9394acdf771b5c393a3447fb and the reasoning for it sounds good 👍 

> Therefore, I do not consider this issue blocking. It'd be nice to resolve, but could be done so in a separate PR later.

I was able to reproduce this issue and tried to figure out why this happens. My understanding is that there is a limitation on how [the `SnackbarSequencer`](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/util/SnackbarSequencer.kt) works were this kind of internal notifications are not really canceled once issued. I agree that this can be investigated further and handled in a follow up PR 👍

Reply:
Thank you for investigating and sharing your insights and perspective. Very helpful! 🙇🏻‍♂️